File: tests/php/testcases/AddStringDictsTest.php
<?
require_once 'base.php';

class AddStringDictsTestCase extends TestServiceTestCase {

    public function test_notset() {
        $p = new ABStringDict();
        $q = new ABStringDict();
        $r = $this->service->add_string_dicts($p, $q);
        $this->assertEquals($r->a, '');
        $this->assertEquals($r->b, '');
    }

    public function test_null() {
        $p = new ABStringDict();
        $p->a = null;
        $p->b = null;
        $q = new ABStringDict();
        $q->a = null;
        $q->b = null;
        $r = $this->service->add_string_dicts($p, $q);
        $this->assertEquals($r->a, '');
        $this->assertEquals($r->b, '');
    }

    public function test_empty() {
        $p = new ABStringDict();
        $p->a = '';
        $p->b = '';
        $q = new ABStringDict();
        $q->a = '';
        $q->b = '';
        $r = $this->service->add_string_dicts($p, $q);
        $this->assertEquals($r->a, '');
        $this->assertEquals($r->b, '');
    }

    public function test() {
        $p = new ABStringDict();
        $p->a = 'hello';
        $p->b = 'say';
        $q = new ABStringDict();
        $q->a = ' world';
        $q->b = ' hello';
        $r = $this->service->add_string_dicts($p, $q);
        $this->assertEquals($r->a, 'hello world');
        $this->assertEquals($r->b, 'say hello');
    }
}

File: tests/mono/testcases/AddStringDictsTest.cs
using System;
using NUnit.Framework;

namespace TestCases {

    [TestFixture()]
    public class AddStringDictsTest : TestServiceTest {

        [Test()]
        public void test_null_args() {
            ABStringDict r = service.add_string_dicts(null, null);
            Assert.AreEqual("", r.a);
            Assert.AreEqual("", r.b);
        }

        [Test()]
        public void test_notset() {
            ABStringDict r = service.add_string_dicts(
                new ABStringDict(), new ABStringDict());
            Assert.AreEqual("", r.a);
            Assert.AreEqual("", r.b);
        }

        [Test()]
        public void test_null() {
            ABStringDict p = new ABStringDict();
            p.a = null;
            p.b = null;
            ABStringDict q = new ABStringDict();
            q.a = null;
            q.b = null;
            ABStringDict r = service.add_string_dicts(p, q);
            Assert.AreEqual("", r.a);
            Assert.AreEqual("", r.b);
        }

        [Test()]
        public void test_empty() {
            ABStringDict p = new ABStringDict();
            p.a = "";
            p.b = "";
            ABStringDict q = new ABStringDict();
            q.a = "";
            q.b = "";
            ABStringDict r = service.add_string_dicts(p, q);
            Assert.AreEqual("", r.a);
            Assert.AreEqual("", r.b);
        }

        [Test()]
        public void test() {
            ABStringDict p = new ABStringDict();
            p.a = "hello ";
            p.b = "say ";
            ABStringDict q = new ABStringDict();
            q.a = "world";
            q.b = "hello";
            ABStringDict r = service.add_string_dicts(p, q);
            Assert.AreEqual("hello world", r.a);
            Assert.AreEqual("say hello", r.b);
        }
    }
}
